lib/resolv.rb

```diff
@@ -116,8 +116,10 @@ def each_address(name)
     yielded = false
     @resolvers.each {|r|
       r.each_address(name) {|address|
-        yield address.to_s
-        yielded = true
+        if block_given?
+          yield address.to_s
+          yielded = true
+        end
```

Comment:
Do you think this change is necessary for SERVFAIL handling? Because calling `each_address` without a block outputs no useful information (it just returns `@resolvers`), it's likely a bug at the call site, and I think ignoring that error is not desirable.

Off-topic: It's conventional in Ruby that an `each_*` method returns an Enumerator when called without a block, which can be implemented as follows (but this is out of the scope of this PR):
```ruby
def each_address(name)
  return enum_for(:each_address, name) unless block_given?
  ...
end
```

Reply:
I reverted the above change, I am not sure why I did it, maybe it was something during my initial testing.

It is my first ruby contribution (and it took me longer to setup my windows environment than fixing ;))